backend/src/folder/folder.resolver.ts

```diff
@@ -0,0 +1,101 @@
+import { Resolver, Query, Mutation, Args, Context } from '@nestjs/graphql';
+import { FolderService } from './folder.service';
+import { Folder } from './entities/folder.entity';
+import { CreateFolderInput } from './dto/create-folder.input';
+import { UpdateFolderInput } from './dto/update-folder.input';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserService } from 'src/user/user.service';
+
+@Resolver(() => Folder)
+export class FolderResolver {
+  constructor(
+    private readonly folderService: FolderService,
+    private readonly prisma: PrismaService,
+    private readonly userService: UserService,
+  ) {}
+
+  @Query(() => [Folder], { nullable: true })
+  async getFolders(@Context() context): Promise<Folder[]> {
+    const authId = context.req.user.sub;
+    const user = await this.userService.getUserByAuthId(authId);
+
+    if (!user) {
+      throw new Error('User not found');
+    }
+    return this.folderService.getFolders(user.id);
```

Comment:
Your getFolders function will always return an array. If there are no folders — it will return an empty array

In this case, this query doesn't need to be nullable

Reply:
I removed nullable